05-html-css/css-grid-intro.md

```diff
@@ -0,0 +1,334 @@
+
+```css
+.container {
+  display: grid;
+  grid-template: 100px 500px / 200px 100px 50%;
+}
+```
+
+This one line is identical to the two properties we saw above.
+
+### The `fr` fractional unit
+
+CSS Grid lets us use one specific unit of measurement that we can't use anywhere else in CSS: the `fr` fractional unit.
+
+Fractional units are relative units: After all other kinds of units are measured, fractional units take the remaining space and calculate their size relative to other fractional units. In other words, they form ratios.
+
+Consider [this Codepen using fractional units](https://codepen.io/adadev/pen/xWKJdz?editors=1100) on different definitions of `grid-template-columns` on different grid containers. Answer the following questions:
+- Which of these examples mixes fractional units and other units?
+- Which of these creates equally sized columns?
+- Which of these creates three columns?
+
+
+## Grid Item
+
+Now we can manipulate our grid items! For the following examples, we'll be using [this Codepen](https://codepen.io/adadev/pen/yKBqQP?editors=1100).
+
+### Using Start and End
+
+We can define on a grid item which column line number the item starts at using `grid-column-start`:
+
+```css
+.container {
+  display: grid;
+  grid-template: 1fr 1fr / 1fr 1fr;
+}
+
+.item {
+  grid-column-start: 1;
+}
+```
+
+Don't forget that CSS Grid's line numbers do not start at "index zero."
+
+We define on a grid item the column line the item ends at using `grid-column-end`:
+
+```css
+.item {
+  grid-column-start: 1;
+  grid-column-end: 4;
+}
+```
+
+We define the start line and end line of an item's row using `grid-row-start` and `grid-row-end`.
+
+```css
+.item {
+  grid-row-start: 3;
+  grid-row-end: 5;
+}
+```
+
+**Practice:** Spend some time uncommenting and altering the `-start` and `-end` properties in our [example Codepen for this section](https://codepen.io/adadev/pen/yKBqQP?editors=1100).
+
+### A shortcut: grid-column and grid-row
+
+You guessed it: there's a shortcut for defining start and end on one line! `grid-column` and `grid-row` can take in both the start value and end value in one line, separated by a slash.
+
+```css
+.item {
+  grid-column: 1 / 4;
+  grid-row: 3 / 5;
+}
+```
+
+### Using `span`
+
+Sometimes you don't know the line number that an item ends at, or you would rather define how many rows or columns that item spans.
+
+Instead of giving an integer for the line number an item ends at, you can give `span num_of_rows_or_columns`.
+
+```css
+.item {
+  grid-column-start: 1;
+  grid-column-end: 4;
+
+  /* is the same as */
+  grid-column-start: 1;
+  grid-column-end: span 3;
+
+  /* is the same as */
+  grid-column: 1 / 4;
+
+  /* is the same as */
+  grid-column: 1 / span 3;
+}
+```
+
+### `auto` in CSS Grid
+
+`auto` means a lot of different things in different contexts. In the context of CSS Grid:
+
```

Comment:
Do we need this section? You don't have any practice for it here, and don't go into it in much detail. Might be more trouble than it's worth.

Reply:
Hm, my only thought is that I end up using it for my solution to the "realistic wireframe" problem